src/pages/preview/[...slug].tsx

```diff
@@ -43,8 +46,16 @@ export default ContentPage;
 export const getServerSideProps = async (context: any) => {
     const { slug } = context.query;
     const slugString = slug.join('/');
-    const contents: ContentItem[] = await getPreviewContent('/' + slugString);
-    const contentItem = contents.filter(c => c.slug === slugString)[0];
+    let contentItem;
+    if (slugString.startsWith('events/')) {
+        contentItem = await getPreviewContentForEvents('/' + slugString);
+    } else {
+        const contents: ContentItem[] = await getPreviewContentForArticles(
+            '/' + slugString
+        );
+        contentItem = contents.filter(c => c.slug === slugString)[0];
```

Comment:
its better to use `.find` here, it returns the entire item that matches the condition.

Reply:
old code. dint touch this, will change.